sparkler-api/src/test/resources/user-agents.txt

```diff
@@ -0,0 +1,19 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# User agents to be used
```

Comment:
> rotating agents should work with any fetcher, wouldn't you agree? 

Agreed - and reflected by the reason why `user-agents.txt` test file is placed inside `sparkler-api/src/test/resources/` directory instead of plugin's directory  :-) 

> if you break it out, and simply make it a part of the core of Sparkler, you won't need it per-fetcher. Does that make sense?

Making part of core Sparkler -- Yes, makes sense. rotating agents and http headers are already part of parent class which is in the core of the system. FYI, the `getUserAgent()` and `this.httpHeaders`  are provided by core system. Here the *per-fetcher* piece of work done is *ask parent class for new agent and set it* before sending the request. For me its seems a reasonable trade off of generalization since each plugin has its own way to fetch resources.